File: SPCode/Security/Support/AspnetUserManager.vb
'Questions? Comments? go to 
'http://www.idesign.net

Imports System
Imports System.Reflection
Imports System.Data.SqlClient
Imports System.Web.Security

Namespace Security
    Friend Class AspNetUserManager
        Implements IUserManager
        Public Function Authenticate(ByVal applicationName As String, ByVal userName As String, ByVal password As String) As Boolean Implements Security.IUserManager.Authenticate
            Membership.ApplicationName = applicationName
            Return Membership.ValidateUser(userName, password)
        End Function
        Public Function IsInRole(ByVal applicationName As String, ByVal userName As String, ByVal role As String) As Boolean Implements Security.IUserManager.IsInRole
            Roles.ApplicationName = applicationName
            Return Roles.IsUserInRole(userName, role)
        End Function
        Public Function GetRoles(ByVal applicationName As String, ByVal userName As String) As String() Implements Security.IUserManager.GetRoles
            Roles.ApplicationName = applicationName
            Return Roles.GetRolesForUser(userName)
        End Function
    End Class
End Namespace

File: SoftLogic.Core/Win/Security/Support/AspnetUserManager.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using System.Reflection;
using System.Web.Security;


namespace SoftLogik.Win.Security
{
	internal class AspNetUserManager : IUserManager
	{
		
		public bool Authenticate(string applicationName, string userName, string password)
		{
			Membership.ApplicationName = applicationName;
			return Membership.ValidateUser(userName, password);
		}
		public bool IsInRole(string applicationName, string userName, string role)
		{
			Roles.ApplicationName = applicationName;
			return Roles.IsUserInRole(userName, role);
		}
		public string[] GetRoles(string applicationName, string userName)
		{
			Roles.ApplicationName = applicationName;
			return Roles.GetRolesForUser(userName);
		}
	}
}
